modules/promotion/src/Plugin/Commerce/PromotionCondition/PromotionConditionBase.php

```diff
@@ -31,4 +33,12 @@ public function execute() {
     return $this->isNegated() ? !$result : $result;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function submitConfigurationForm(array &$form, FormStateInterface $form_state) {
+    // @todo Update plugin_select to pass subform state automatically.
```

Comment:
Commerce plugins all rely on the complete form state. It would be weird to make conditions an exception to that, so the current workaround sounds better.